rust-src/id/src/identity_provider.rs

```diff
@@ -246,13 +246,462 @@ pub fn validate_request<P: Pairing, C: Curve<Scalar = P::ScalarField>>(
     }
 }
 
+pub fn validate_request<P: Pairing, C: Curve<Scalar = P::ScalarField>>(
+    pre_id_obj: &PreIdentityObject<P, C>,
+    context: IpContext<P, C>,
+) -> Result<(), Reason> {
+    let pub_info_for_ip = &pre_id_obj.pub_info_for_ip;
+    let id_cred_pub = &pre_id_obj.pub_info_for_ip.id_cred_pub;
+    let common_fields = &pre_id_obj.common_fields;
+    let poks_common = &pre_id_obj.poks.common_proof_fields;
+    let proof_acc_sk = &pre_id_obj.poks.proof_acc_sk;
+    // Verify signature:
+    let keys = &pub_info_for_ip.vk_acc.keys;
+    let threshold = pub_info_for_ip.vk_acc.threshold;
+
+    // message signed
+    let signed = Sha256::digest(&to_bytes(&pub_info_for_ip));
+
+    let reason = Reason::IncorrectProof; // TODO: introduce different reason
```

Comment:
Why do we define this constant? I think it would be better to just use `Reason::IncorrectProof` where this value is used. It is clearer. And I don't understand what the TODO is for.